src/implementations/vanilla/src/basics/form-elements/text-field/text-field.js

```diff
@@ -38,6 +38,8 @@ class TextField extends Core {
         if (this.initialOptions && this.initialOptions.icon) {
             this.setIcon(this.initialOptions.icon);
         }
+        this._findDOMEl('.hig__text-field__clear-button')
+            .addEventListener('click', () => this._clearValue());
```

Comment:
This event listener is going to get us into trouble when we want to control the text field's value from React. 

As a guiding principal, the vanilla layer exposes an API that allows our React layer (or other theoretical wrappers) to put a component into each of its visual states, as well as attach event listeners. We generally do not enable interactivity in the vanilla layer.

For this story I'd make `showClearButton` and `hideClearButton` public, and add `onClearButtonClick`.